### PR TITLE
refactor: enable additional lints

### PR DESCRIPTION
**Pull request by CBenoit**
I took the list that we used for IronRDP and removed a few that I believed didn’t apply.

Can you review this? @pacmancoder

### CODE REVIEW

File: crates/network-scanner-net/examples/basic_ping.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![allow(unused_crate_dependencies)]
```
Comment (pacmancoder) [inline crates/network-scanner-net/examples/basic_ping.rs:1]: Maybe let's create a separate issue to address removing these kinds of lints later?

File: crates/network-scanner/src/broadcast/blocking.rs [hunk 2]
```diff
@@ -34,6 +34,7 @@ impl Iterator for BorcastBlockStream {
             return None;
         }
 
+        // SAFETY: TODO: explain safety.
```
Comment (pacmancoder) [inline crates/network-scanner/src/broadcast/blocking.rs:37]: Let's add this to follow-up issue

**Review by pacmancoder — APPROVED** [review]
LGTM overall, although I think we need to address unused crate dependencies and TODOs in `SAFETY` comments in the follow-up issue/PR

File: crates/network-scanner-net/examples/basic_ping.rs (hunk 1 above)
Comment (CBenoit) [reply crates/network-scanner-net/examples/basic_ping.rs:1]: Agreed, my intention is to send another PR to rework the workspace. I’ll follow up on this soonish.

**Pull request merged**